app/scripts/lib/account-tracker.js

```diff
@@ -336,26 +335,29 @@ export default class AccountTracker {
    */
   async _updateAccountsViaBalanceChecker(addresses, deployedContractAddress) {
     const { accounts } = this.store.getState();
-    this.web3.setProvider(this._provider);
-    const ethContract = this.web3.eth
-      .contract(SINGLE_CALL_BALANCES_ABI)
-      .at(deployedContractAddress);
-    const ethBalance = ['0x0'];
-
-    ethContract.balances(addresses, ethBalance, (error, result) => {
-      if (error) {
-        log.warn(
-          `MetaMask - Account Tracker single call balance fetch failed`,
-          error,
-        );
-        Promise.all(addresses.map(this._updateAccount.bind(this)));
-        return;
-      }
+    this.ethersProvider = new ethers.providers.Web3Provider(this._provider);
```

Comment:
Why do we need to set this again? Isn't it already set on line 87? I see that you're replicating the logic from the web3 implementation... I will try without this line included? I can understand if we need to update the provider when the network changes, but in this current implementation `this._provider` doesn't appear to change...

Reply:
hmmm actually hold up on this. Looking through it and seeing whether this is indeed necessary.

Reply:
Ok so it does appear to be necessary in this case... I'm just not yet sure how `this._provider` is getting updated without getting reset

Reply:
But this change should stay @amerkadicE